src/util.py

```diff
@@ -260,12 +265,86 @@ def xr_add_cyclic_longitudes(da, coord):
 
     return new_da
 
+
+def set_vector_density(data, minDistance=0, otherVars=[]):
+    """
+    Utility function to change density of vector plots.
+
+    Ars:
+
+        data (:class:`xarray.core.dataarray.DataArray`):
+            Data array that contains the vector plot latitude/longitude data.
+
+        minDistance (:class:`int`):
+            Value in degrees that determines the distance between the vectors.
+
+        otherVars (:class: array of `xarray.core.dataarray.DataArray`):
+            Other variables that require the same density reduction as the lat/lon data
+
+    """
+    import math
+    import warnings
+
+    if minDistance <= 0:
+        raise Exception("minDistance cannot be negative or zero.")
+    else:
+        lat_every = 1
+        lon_every = 1
+
+        # Get distance between points in latitude (y axis)
+        lat = data['lat']
+        latdifference = (float)(lat[1] - lat[0])
+
+        # Get distance between points in longitude (x axis)
+        lon = data['lon']
+        londifference = (float)(lon[1] - lon[0])
+
+        # Get distance between points that are diagonally adjacent
+        diagDifference = math.sqrt(latdifference**2 + londifference**2)
+
+        # Initialize ds
+        ds = data
+
+        if diagDifference >= minDistance and latdifference >= minDistance and londifference >= minDistance:
+            warnings.warn('Plot spacing is alrady greater or equal to ' + (str)(minDistance))
+
+        # While diagD
+        while diagDifference < minDistance or latdifference < minDistance or londifference < minDistance:
+
+            # Get distance between points in latitude (y axis)
+            lat = data['lat']
+            latdifference = (float)(lat[lat_every] - lat[0])
+
+            # Get distance between points in longitude (x axis)
+            lon = data['lon']
+            londifference = (float)(lon[lon_every] - lon[0])
+
+            # Get distance between points that are diagonally adjacent
+            diagDifference = math.sqrt(latdifference**2 + londifference**2)
+
+            lat_every += 1
+            lon_every += 1
+
+            ds = data.isel(lat=slice(None, None, lat_every), lon=slice(None, None, lon_every))
+
+        lon_size = data['lon'].size
+        lat_size = data['lat'].size
+
+        for variable in range(len(otherVars)):
+            otherVars[variable] = otherVars[variable][0:lat_size:lat_every, 0:lon_size:lon_every]
```

Comment:
I have a few thoughts here about `otherVars`:

1. It seems kind of mysterious to me what other variables from the data set might be `otherVars`. May be give some examples or further explanation in `docstrings`, if possible.

2. Is it intentional to slice them in a more numpy-ic way than the xArray slicing methods like you already do in the above `while-loop` for `lat` and `lon`? It makes me doubted if `otherVars` are numpy arrays despite what is said in the `docstrings`?

Reply:
I just tested this some more times, and otherVars was only meant to slice the variables that weren't lat/lon (such as U and V) to make sure that all of the variables being used in the quiver function had the same dimensions. Apparently, that is done automatically when you slice lat/lon, so the otherVars argument isn't even necessary.